src/main/java/seedu/sudohr/logic/parser/ParserUtil.java

```diff
@@ -164,6 +168,45 @@ public static LeaveDate parseLeaveDate(String date) throws ParseException {
         return new LeaveDate(leaveDate);
     }
 
+    /**
+     * Parses a {@code String startDate} and {@code String endDate} into a list of {@code leaveDates}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
+     * @throws ParseException if start date is more than 7 days earlier than end date
+     */
+    public static List<LeaveDate> parseLeaveDateFromTo(String startDateString, String endDateString)
+            throws ParseException {
+        requireAllNonNull(startDateString, endDateString);
+        String trimmedStartDate = startDateString.trim();
+        String trimmedEndDate = endDateString.trim();
+        LocalDate startDay;
+        LocalDate endDay;
+        try {
+            startDay = LocalDate.parse(trimmedStartDate);
+            endDay = LocalDate.parse(trimmedEndDate);
+        } catch (DateTimeParseException e) {
+            throw new ParseException(LeaveDate.MESSAGE_CONSTRAINTS);
+        }
+
+        if (startDay.isAfter(endDay)) {
+            throw new ParseException(AddEmployeeToLeaveFromToCommand.DATE_CONSTRAINTS);
+        }
+
+        if (startDay.until(endDay).getDays() >= 7) {
+            throw new ParseException(AddEmployeeToLeaveFromToCommand.DATE_CONSTRAINTS);
+        }
```

Comment:
can be merged into a single conditional if you want (eg. using || operator)

Reply:
I prefer not to to make code easier to read